client/landing/subscriptions/hooks/use-recommended-site.ts

```diff
@@ -44,73 +35,39 @@ export const useRecommendedSite = (
 	} );
 
 	// Memoized selector to check if item is in recommended list
-	// Try matching by feedId first (since that's what we add to the list now), fall back to siteId
 	const selectIsInRecommendedList = useCallback(
 		( state: AppState ) => {
 			if ( ! currentUserName || ! recommendedBlogsList?.ID ) {
 				return false;
 			}
 
-			// Try to match by feedId first (preferred, since that's what we add to the list)
+			// Match by feedId only
 			const matchByFeedId = getMatchingItem( state, { listId: recommendedBlogsList.ID, feedId } );
-			if ( matchByFeedId ) {
-				return true;
-			}
-
-			// Fall back to matching by siteId for backward compatibility
-			if ( blogId ) {
-				const matchBySiteId = getMatchingItem( state, {
-					listId: recommendedBlogsList.ID,
-					siteId: blogId,
-				} );
-				return !! matchBySiteId;
-			}
-
-			return false;
+			return !! matchByFeedId;
 		},
-		[ currentUserName, recommendedBlogsList?.ID, feedId, blogId ]
+		[ currentUserName, recommendedBlogsList?.ID, feedId ]
 	);
 
-	// Get actual state from Redux
-	const isInRecommendedList = useSelector( selectIsInRecommendedList );
+	// Get actual state from Redux (includes optimistic updates handled by reducer)
+	const isRecommended = useSelector( selectIsInRecommendedList );
 
-	// Local state for optimistic updates and loading state
-	const [ optimisticRecommendedState, setOptimisticRecommendedState ] = useState< boolean | null >(
-		null
-	);
+	// Local state for loading state only
 	const [ isUpdating, setIsUpdating ] = useState( false );
 
-	// Use optimistic state if available, otherwise fall back to selector value
-	const isRecommended =
-		optimisticRecommendedState !== null ? optimisticRecommendedState : isInRecommendedList;
-
 	// Determine if toggle is available
 	const canToggle = Boolean(
 		currentUserName && typeof currentUserName === 'string' && recommendedBlogsList?.ID
 	);
 
-	// Sync optimistic state with selector when selector changes
-	useEffect( () => {
-		if (
-			optimisticRecommendedState !== null &&
-			optimisticRecommendedState === isInRecommendedList
-		) {
-			// Optimistic state matches actual state, clear optimistic state
-			setOptimisticRecommendedState( null );
-		}
-	}, [ isInRecommendedList, optimisticRecommendedState ] );
-
-	// Toggle function with optimistic updates
+	// Toggle function - Redux reducer handles optimistic updates
 	const toggleRecommended = useCallback( () => {
```

Comment:
Should we would want to debounce this toggle to prevent hitting it back and forth rapidly?